lib/sequelize.js

```diff
@@ -1239,6 +1239,9 @@ Sequelize.prototype.transaction = function(options, autoCallback) {
         // If an error occured while commiting a transaction do not try to rollback the transaction
         if (transaction.finished === 'commit') {
           reject(err);
+        } else if (transaction.finished === 'rollback'){
+          // Already in rollback state. Reject with the original error.
+          reject(err);
```

Comment:
Sounds fine to me. I didn't know if there may be other transaction states to handle. 

Should I update the PR?